packages/spec/src/error.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright IBM Corp. 2019, 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import format from 'pretty-format';
+
+export class ValidationError extends Error {
```

Comment:
I know we have ran in to issues with `chai` not supporting "warnings" which is a great feature and one we've been looking to add, by any chance have you looked in to if Jest expect supports warnings versus having to maintain a custom assertion class?  We have a spike to look in to this work but have not been able to prioritize it at this point.

Reply:
In this case, the vision is that `ValidationError` would be able to output context-sensitive error messages. It currently does not operate that way, but the thinking could be:

```
new ValidationError([
  'description of the error',
  // node where the error occurs
  // root node
]);
```

And it could spit out a diff of expected versus actual for the node and tree in question.